src/include/Context.h

```diff
@@ -463,6 +464,19 @@ class FunctionContext : public Context {
   boost::function<void(int)> m_callback;
 };
 
+class LambdaContext : public Context {
```

Comment:
Minor: can we hold off on this for now and just use FunctionContext which also takes lambda?  I'd rather see a cleanup ticket to switch from FunctionContext / boost::function to the C++11 native.